src/applications/vaos/tests/new-appointment/components/ReasonForAppointmentPage.unit.spec.jsx

```diff
@@ -1,189 +1,236 @@
+    vaOnlineSchedulingDirect: true,
+    vaOnlineSchedulingCommunityCare: true,
+  },
+  user: {
+    profile: {
+      facilities: [
+        { facilityId: '983', isCerner: false },
+        { facilityId: '984', isCerner: false },
+      ],
+    },
+  },
+};
+
+describe('VAOS integration: reason for appointment page with a single-site user', () => {
+  beforeEach(() => mockFetch());
+  afterEach(() => resetFetch());
+
+  it('should show page for VA medical request', async () => {
+    const store = createTestStore(initialState);
+    const screen = renderWithStoreAndRouter(<ReasonForAppointmentPage />, {
+      store,
+    });
+
+    expect(screen.baseElement).to.contain.text(
+      'Please let us know why you’re making this appointment',
+    );
 
-describe('VAOS <ReasonForAppointmentPage>', () => {
-  it('should render', () => {
-    const openReasonForAppointment = sinon.spy();
-    const updateReasonForAppointmentData = sinon.spy();
+    expect(screen.getAllByRole('radio').length).to.equal(4);
 
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        updateReasonForAppointmentData={updateReasonForAppointmentData}
-        data={{}}
-      />,
+    expect(
+      screen.getByRole('heading', {
+        name: /If you have an urgent medical need, please:/i,
+      }),
     );
-
-    expect(form.find('input').length).to.equal(4);
-    expect(form.find('textarea').length).to.equal(1);
-    form.unmount();
   });
 
-  it('should not submit empty form', () => {
-    const openReasonForAppointment = sinon.spy();
-    const history = {
-      push: sinon.spy(),
-    };
+  it('should show page for Community Care medical request', async () => {
+    const store = createTestStore(initialState);
+    await setTypeOfFacility(store, /Community Care/i);
 
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        history={history}
-        data={{}}
-      />,
-    );
+    const screen = renderWithStoreAndRouter(<ReasonForAppointmentPage />, {
+      store,
+    });
 
-    form.find('form').simulate('submit');
+    expect(screen.baseElement).to.contain.text(
+      'Tell us the reason for this appointment',
+    );
 
-    expect(form.find('AlertBox').length).to.equal(1);
+    expect(screen.getByRole('textbox')).to.exist;
 
-    expect(form.find('.usa-input-error').length).to.equal(2);
-    expect(history.push.called).to.be.false;
-    form.unmount();
+    expect(
+      screen.getByRole('heading', {
+        name: /If you have an urgent medical need, please:/i,
+      }),
+    );
   });
 
-  it('should call updateReasonForAppointmentData after change', () => {
-    const openReasonForAppointment = sinon.spy();
-    const updateReasonForAppointmentData = sinon.spy();
-    const history = {
-      push: sinon.spy(),
-    };
+  it('should show validation for VA medical request', async () => {
```

Comment:
One other thing to note with these validation messages is that there's some hidden functionality as far as what the character limit for the reason field is. If you get to this page as part of the direct schedule flow, then the character limit changes depending on the radio option that you choose (because we concatenate that value and the reason info into one field to send to the backend).

I think the easiest way to set the flow would be to dispatch this action: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/new-appointment/redux/actions.js#L197

So like `store.dispatch(startDirectSchedulFlow())` in your test before the render.

Reply:
Gotcha, implemented the dispatch in the beforeEach call.